src/variables.jl

```diff
@@ -1227,6 +1227,21 @@ function add_variable(
     return reshape_vector(var_refs, variable.shape)
 end
 
+function add_variable(
+    model::Model,
+    variable::VariablesConstrainedOnCreation,
+    name::String,
+)
+    var_indices = _moi_add_constrained_variables(
+        backend(model),
+        variable.scalar_variables,
+        variable.set,
+        fill(name, length(variable.scalar_variables)),
```

Comment:
If the purpose of https://github.com/jump-dev/JuMP.jl/pull/2978/ was performance, maybe we don't want to create this dummy array identical strings ? Either using FillArray or adding support for it in `_moi_add_constrained_variables`.

Reply:
The performance problem was passing the names to solvers, not this allocation.

Reply:
Okay, I got rid of this.